shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts

```diff
@@ -181,4 +196,19 @@ export class BaseCore implements IAppInsightsCore {
 
         return new ProcessTelemetryContext(thePlugins, _this.config, _this);
     }
+
+    areExtensionsInitialized(): boolean {
+        let _this = this;
+        if (!_this._allExtensionsInitialized) {
+            let extensions = _this._extensions;
+            let isAnyExtensionNotInit = false;
+            _arrForEach(extensions, (ext: ITelemetryPlugin) => {
```

Comment:
Why is a foreach used? Does it compress better?

Reply:
I don't believe it compress better than a regular for, I don't have a particular reason to use it, is there any issue with it?

Reply:
No issue with it, I think a `for...of` would make it not require the temp return var and be easier to read.